public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/reducer.ts

```diff
@@ -78,13 +79,38 @@ export const queriesAndExpressionsReducer = createReducer(initialState, (builder
     })
     .addCase(updateExpression, (state, { payload }) => {
       state.queries = state.queries.map((query) => {
-        return query.refId === payload.refId
-          ? {
-              ...query,
-              model: payload,
-            }
-          : query;
+        const dataSource = state.queries.find((alertQuery) => alertQuery.refId === payload.expression);
```

Comment:
I would rename `dataSource` to `query`. 
I think there is one more edge case. What if the resample input is not a query but a different expression?

Reply:
💯, I will take a look on this edge case @konrad147